File: result/translate/chatgpt/java2python/python/REMOVE_MINIMUM_ELEMENTS_EITHER_SIDE_2MIN_MAX.py
def f_gold(arr, n):
    longest_start = -1
    longest_end = 0
    
    for start in range(n):
        min_val = float('inf')
        max_val = float('-inf')
        
        for end in range(start, n):
            val = arr[end]
            
            if val < min_val:
                min_val = val
            if val > max_val:
                max_val = val
                
            if 2 * min_val <= max_val:
                break
                
            if end - start > longest_end - longest_start or longest_start == -1:
                longest_start = start
                longest_end = end
                
    if longest_start == -1:
        return n
    
    return n - (longest_end - longest_start + 1)

File: result/translate/starchat/python2java/java/REMOVE_MINIMUM_ELEMENTS_EITHER_SIDE_2MIN_MAX.java
public static int f_gold ( int arr [ ], int n ) {
        int longest_start = -1 ;
        int longest_end = 0 ;
        for ( int start = 0 ; start < n ; start++ ) {
            int min = Integer.MAX_VALUE ;
            int max = Integer.MIN_VALUE ;
            for ( int end = start ; end < n ; end++ ) {
                int val = arr [ end ] ;
                if ( val < min ) min = val ;
                if ( val > max ) max = val ;
                if ( 2 * min <= max ) break ;
                if ( end - start > longest_end - longest_start || longest_start == -1 ) {
                    longest_start = start ;
                    longest_end = end ;
                }
            }
        }
        if ( longest_start == -1 ) return n ;
        return n - ( longest_end - longest_start + 1 ) ;
    }